File: Java/Longest Common Prefix in an Array.java

class Solution{
    String longestCommonPrefix(String arr[], int n){
        // code here
        StringBuilder sb = new StringBuilder();
        int min = arr[0].length();
        for(int i=0; i<n; i++) min = Math.min(min,arr[i].length());
        boolean flag = true;
        int j = 0;
        while(j<min){
            for(int i=1; i<n; i++){
                if(arr[i-1].charAt(j)!=arr[i].charAt(j)){
                    flag = false;
                    break;
                }
            }
            if(flag==false) break;
            sb.append(arr[0].charAt(j++));
        }
        if(sb.length()==0) return "-1";
        return sb.toString();
    }
}

File: cpp/Longest Common Prefix in an Array.cpp
class Solution{
  public:
    
    string longestCommonPrefix (string arr[], int N)
    {
        // your code here
        int len = arr[0].length();
        string str = "";
        
        for(int i=0; i<N; i++){
            len = arr[i].length()<len?arr[i].length():len;
        }
        
        int j = 0;
        bool flag = true;
        
        while(j<len){
            
            for(int i=1; i<N; i++){
                if(arr[i-1][j]!=arr[i][j]) {
                    flag = false;
                    break;
                }
            }
            if(!flag) break;
            str = str + arr[0][j++];
        }
        if(str.length()==0) return "-1";
        return str;
    }
};